cin_validator/rules/cin2022_23/rule_8805.py

```diff
@@ -0,0 +1,166 @@
+    # replace ChildIdentifiers with the value in the module column of the excel sheet corresponding to this rule .
+    module=CINTable.CINdetails,
+    # replace the message with the corresponding value for this rule, gotten from the excel sheet.
+    message="A CIN case cannot have a CIN closure date without a Reason for Closure",
+    # The column names tend to be the words within the < > signs in the github issue description.
+    affected_fields=[CINclosureDate, ReasonForClosure],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # PREPARING DATA
+
+    # Replace ChildIdentifiers with the name of the table you need.
+    df = data_container[CINDetails]
+    # Before you begin, rename the index so that the initial row positions can be kept intact.
+    df.index.name = "ROW_ID"
+
+    # lOGIC
+    # Implement rule logic as described by the Github issue.
+    # Put the description as a comment above the implementation as shown.
+
+    # If <CINclosureDate> (N00102) is present then <ReasonForClosure> (N00103) must also be present
+    # return rows where CINClosureDate is present but ReasonForClosure is not.
+    # If CINclosureDate is not null and ReasonForClosure is null
+    condition = df[CINclosureDate].notna() & df[ReasonForClosure].isna()
+
+    # get all the data that fits the failing condition. Reset the index so that ROW_ID now becomes a column of df
+    df_issues = df[condition].reset_index()
+
+    # SUBMIT ERRORS
+    # Generate a unique ID for each instance of an error. In this case,
+    # - If only LAchildID is used as an identifier, multiple instances of the error on a child will be understood as 1 instance.
+    # We don't want that because in reality, a child can have multiple instances of an error.
+    # - If we use the LAchildID-CPPstartDate combination, that artificially cancels out the instances where a start date repeats for the same child.
+    # Another rule checks for that condition. Not this one.
+    # - It is very unlikely that a combination of LAchildID-CPPstartDate-CPPendDate will repeat in the DataFrame.
+    # Hence, it can be used as a unique identifier of the row.
+
+    # Replace CPPstartDate and CPPendDate below with the columns concerned in your rule.
+    link_id = tuple(
+        zip(
+            df_issues[LAchildID], df_issues[CINclosureDate], df_issues[ReasonForClosure]
+        )
+    )
+    df_issues["ERROR_ID"] = link_id
+    df_issues = df_issues.groupby("ERROR_ID")["ROW_ID"].apply(list).reset_index()
+    # Ensure that you do not change the ROW_ID, and ERROR_ID column names which are shown above. They are keywords in this project.
+    rule_context.push_type_1(
+        table=CINDetails, columns=[CINclosureDate, ReasonForClosure], row_df=df_issues
+    )
+
+
+def test_validate():
+    # Create some sample data such that some values pass the validation and some fail.
+    fake_date_frame = pd.DataFrame(
+        [
+            {
```

Comment:
Good work on these, I just deleted a few extra lines of code that were commented out that made the code a little less tidy!